website/src/controllers/api-v3/user.js

```diff
@@ -41,6 +45,47 @@ api.getUser = {
   },
 };
 
+/**
+ * @api {delete} /user DELETE an authenticated user's profile
+ * @apiVersion 3.0.0
+ * @apiName UserDelete
+ * @apiGroup User
+ *
+ * @apiSuccess {} object An empty object
+ */
+api.deleteUser = {
+  method: 'DELETE',
+  middlewares: [authWithHeaders(), cron],
+  url: '/user',
+  async handler (req, res) {
+    let user = res.locals.user;
+    let plan = user.purchased.plan;
+
+    if (plan && plan.customerId && !plan.dateTerminated) {
+      throw new NotAuthorized(res.t('cannotDeleteActiveAccount'));
+    }
+
+    let types = ['party', 'publicGuilds', 'privateGuilds'];
+    // @TODO: The group leave route doesn't work unless it has these fields. We should probably force the group to get these
+    let groupFields = basicGroupFields.concat(' leader memberCount');
```

Comment:
It took some tinkering, but I discovered that you need to populate the leader field for this check to work: https://github.com/HabitRPG/habitrpg/blob/api-v3/website/src/models/group.js#L613 .

It is the same for memberCount and this check: https://github.com/HabitRPG/habitrpg/blob/api-v3/website/src/models/group.js#L597 .

I think it should be the model responsibility to enforce the fields are available for checking. I'm not exactly sure how that would work though.